ethcore/src/client/client.rs

```diff
@@ -2217,22 +2217,22 @@ impl BlockChainClient for Client {
 }
 
 impl IoClient for Client {
-	fn queue_transactions(&self, transactions: Vec<Bytes>, peer_id: usize) {
+	fn queue_transactions(&self, transactions: Vec<UnverifiedTransaction>, peer_id: usize) {
 		trace_time!("queue_transactions");
 		let len = transactions.len();
 		self.queue_transactions.queue(&self.io_channel.read(), len, move |client| {
 			trace_time!("import_queued_transactions");
-
-			let txs: Vec<UnverifiedTransaction> = transactions
-				.iter()
-				.filter_map(|bytes| client.engine.decode_transaction(bytes).ok())
-				.collect();
-
+			let hashes: Vec<_> = transactions.iter().map(UnverifiedTransaction::hash).collect();
 			client.notify(|notify| {
-				notify.transactions_received(&txs, peer_id);
+				notify.transactions_received(&hashes, peer_id);
```

Comment:
maybe we should move this notification after miner imports the transactions? cause those transactions may be invalid

Reply:
AFAIR this notification only updates `last_sent_hashes` of a peer, so that we don't propagate the same transaction that the peer sent us